src/mito2/src/error.rs

```diff
@@ -669,7 +671,7 @@ impl ErrorExt for Error {
             CompactRegion { source, .. } => source.status_code(),
             CompatReader { .. } => StatusCode::Unexpected,
             InvalidRegionRequest { source, .. } => source.status_code(),
-            RegionReadonly { .. } => StatusCode::RegionReadonly,
+            RegionState { .. } => StatusCode::RegionReadonly,
```

Comment:
Should we rename the status code?

Reply:
I think we should remove it, and use `RegionNotReady` here.

Reply:
There is also a state `RegionBusy`, I'm going to use it. cc @WenyXu Do we rely on the `RegionReadonly` state?

Reply:
Well, maybe `RegionNotReady` is proper.